ion-hash/tests/ion_hash_tests.rs

```diff
@@ -1,12 +1,109 @@
 // Copyright Amazon.com, Inc. or its affiliates.
 
-use digest::Digest;
+use digest::consts::U4096;
+use digest::{FixedOutput, Reset, Update};
 use ion_hash;
 use ion_rs::result::{illegal_operation, IonResult};
 use ion_rs::value::reader::{element_reader, ElementReader};
 use ion_rs::value::*;
-use sha2::Sha256;
 use std::fs::read;
+use std::iter::FromIterator;
+
+#[derive(Default, Clone)]
+struct IdentityDigest(Vec<u8>);
+
+impl IdentityDigest {
+    fn output_size() -> usize {
+        4096usize
+    }
+}
+
+impl Update for IdentityDigest {
+    fn update(&mut self, data: impl AsRef<[u8]>) {
+        // When we we get the finalized output of another `IdentityDigest`, it's
+        // going to have a ton of trailing zeros. Sometimes Ion Hash actually
+        // does call update with trailing zeros (e.g. float negative zero). So
+        // we have a problem: when should we use the data verbatim and when
+        // should we strip the trailing zeros that are *only there because of
+        // the fixed output trait*? The hack is to strip trailing zeroes if we
+        // get called with something that looks like it came from the result of
+        // another IdentityDigest. Note that `>=` is used here due to escaping
+        // (which ironically, means the output isn't fixed size).
+        let data = data.as_ref();
+        let actual_update = if data.len() >= IdentityDigest::output_size() {
+            without_trailing_zeros(data)
```

Comment:
😅

Reply:
Oof.. yeah this is, pretty special.  I thought you were going to use SHA2 to derive the tests from identity?  Is that in a later PR?  If so doesn't that solve this problem all together?

Reply:
I had to undo that because it's nearly impossible to get right with subhashers

Reply:
btw I explained this is the commit message :) not sure if you saw that.

Reply:
Re-reading the PR summary, yes, that explains it...

I think we could have the test data be more structural to make this doable, but that is a larger change so I get trying to make this work.

Reply:
So does that mean you're OK with the code as is, or should I make some changes?

Reply:
I am okay, but not particularly happy with it.  I mean really we're fighting the testing file and the way the `Digest` trait works.  I think this warrants a TODO around making these tests not wag the dog, but I don't think that belongs in this PR.